### PR TITLE
Add a Stdlib transform from Input.t to TTFI Bam

### PR DESCRIPTION
**Pull request by ihodes**

### CODE REVIEW

File: src/pipeline_edsl/pipeline_library.mli [hunk 1]
```diff
@@ -4,15 +4,25 @@ module Input : sig
...
   and fastq_fragment = (string option * fastq_data)
   and fastq = {
-    sample_name : string;
+    fastq_sample_name : string;
     files : fastq_fragment list;
   }
+  and bam = {
+    bam_sample_name: string;
+    path: string;
+    how: [ `PE | `SE ];
```
Comment (smondet) [inline src/pipeline_edsl/pipeline_library.mli]: You didn't use the `how` here

File: src/pipeline_edsl/pipeline_library.mli [hunk 2]
```diff
@@ -4,15 +4,25 @@ module Input : sig
...
+    sorting: sorting option;
+    reference_build: string;
+  }
   and fastq_data =
     | PE of string * string
     | SE of string
     | Of_bam of [ `PE | `SE ] * [ `Coordinate | `Read_name ] option * string * string
```
Comment (smondet) [inline src/pipeline_edsl/pipeline_library.mli]: Also this looks like it should be factored, at least using the `how` and `sorting` types :)

File: src/pipeline_edsl/pipeline_library.ml [hunk 3]
```diff
@@ -7,22 +7,34 @@ open Biokepi_run_environment.Common
...
   and fastq = {
-    sample_name : string;
-    files : fastq_fragment list; (* TODO: rename to “fragments” *)
+    fastq_sample_name : string;
+    files : fastq_fragment list;
+  }
+  and bam = {
+    bam_sample_name: string;
+    path: string;
+    how: [ `PE | `SE ];
```
Comment (smondet) [inline src/pipeline_edsl/pipeline_library.ml]: Here also the `how` should be used.

**Comment by ihodes** [comment]
Thanks for the review; addressed comments & back to you!

**Pull request merged**